### PR TITLE
errata-print02-ch02-lst03: superfluous punctuation

### LINKED ISSUE #39: Possible superfluous `&*`: print02-ch02-lst03
Not submitting this as an errata just yet because it could be my misunderstanding.

Listings 2-2 and 2-3 contrast static vs dynamic dispatch.

2-2:
```rust
impl String {
    pub fn contains(&self, p: impl Pattern) -> bool {
        p.is_contained_in(self)
    }
}
```

2-3:
```rust
impl String {
    pub fn contains(&self, p: &dyn Pattern) -> bool {
        p.is_contained_in(&*self)
    }
}
```

I was curious what the purpose of `&*` in listing 2-3 was, especially since it's not present in listing 2-2 and I don't know why the explicit dereference would be necessary for dynamic but not for static dispatch (especially the dispatch isn't happening on `self` but on `p`). I wrote up 2-3 [in the Playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=121153a798e90f33bd661db58944ec99), _without_ the `&*` (and also modified to not conflict with the standard library types), and it compiled, so it seems like in fact it's not necessary.

I did notice that the string argument for `Pattern::is_contained_in` is `&str`, so there does need to be a Deref conversion, but `&self` should already do this, whether or not `is_contained_in` is dynamically or statically dispatched.

Even if this particular example doesn't need `&*`, is there a situation in which switching some value from an `impl` to a `dyn` type would require introducing `&*`?

### PR DESCRIPTION
**Pull request by BatmanAoD**
(Sorry about the branch name; I forgot to modify it.)

Resolves #39.

### CODE REVIEW

File: _errata/errata-print02-ch02-lst03-superfluous-punctuation [hunk 1]
```diff
@@ -0,0 +1,30 @@
+---
+printing: second
```
Comment (jonhoo) [inline _errata/errata-print02-ch02-lst03-superfluous-punctuation]: No need for this — the error is present in the first printing as well as far as I can tell. Same with the file name :+1:

File: _errata/errata-print02-ch02-lst03-superfluous-punctuation [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+```rust
+impl String {
+    pub fn contains(&self, p: &dyn Pattern) -> bool {
+        p.is_contained_in(self)
+    }
+}
+```
+
+Resolves #39
```
Comment (jonhoo) [inline _errata/errata-print02-ch02-lst03-superfluous-punctuation]: This should just be in the Git commit message, and doesn't need to be in the Markdown file.

**Comment by BatmanAoD** [comment]
@jonhoo Fixed!

**Comment by jonhoo** [comment]
Ah, can you keep `print01` in the file name too (to match the other errata)?

**Comment by BatmanAoD** [comment]
@jonhoo `print01` or `print02`? Or both?

**Comment by jonhoo** [comment]
print01 is fine 👍

**Comment by BatmanAoD** [comment]
done

**Comment by jonhoo** [comment]
Oh, the file name is now missing a `.md`!

**Comment by BatmanAoD** [comment]
That's...odd... fixed

**Review by jonhoo — APPROVED** [review]

**Pull request merged**